File: jules-scratch/verification/verify_gallery.py
import asyncio
from playwright.async_api import async_playwright

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        page = await browser.new_page()

        await page.goto("http://localhost:8000/index.html", wait_until="domcontentloaded")

        # Give the particles some time to appear
        await page.wait_for_timeout(10000)

        await page.screenshot(path="jules-scratch/verification/gallery_with_particles.png")

        await browser.close()

if __name__ == "__main__":
    asyncio.run(main())

File: verification/verify_gallery.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  try {
    console.log('Navigating to http://localhost:8080');
    await page.goto('http://localhost:8080');

    console.log('Waiting for .model-card');
    await page.waitForSelector('.model-card', { timeout: 60000 });

    // Wait a bit for images/models to render
    await page.waitForTimeout(5000);

    console.log('Taking screenshot');
    await page.screenshot({ path: 'verification/gallery_real.png' });
    console.log('Screenshot saved to verification/gallery_real.png');
  } catch (e) {
    console.error(e);
    process.exit(1);
  } finally {
    await browser.close();
  }
})();
